src/core/utils/number/numberUtils.ts

```diff
@@ -0,0 +1,80 @@
+function formatNumber(
+  providedOptions: Omit<Intl.NumberFormatOptions, "style"> & {
+    locale?: string;
+  } = {}
+) {
+  const {locale, ...otherOptions} = providedOptions;
+  const options = {
+    style: "decimal",
+    ...otherOptions
+  };
+
+  let numberFormatter: {
+    format: (x: number | bigint) => string;
+  };
+
+  try {
+    numberFormatter = new Intl.NumberFormat(
+      locale || [navigator.language, "en-GB"],
+      options
+    );
+  } catch (error) {
+    numberFormatter = {
+      format(x: number | bigint) {
+        return x.toLocaleString(locale);
+      }
+    };
+  }
+
+  return (value: number) => {
+    let formattedValue = "";
+
+    if (!Object.is(value, NaN)) {
+      formattedValue = numberFormatter.format(value);
+    }
+
+    return formattedValue;
+  };
+}
+
+/**
+ * Coerces a number scientific notation. {@link https://observablehq.com/@mbostock/localized-number-parsing | Reference}
+ * @param {number} value - A number to convert to string
+ * @param {string} locale - Default locale used is browser locale
+ * @returns {string} The value after coercing the given value to a scientific notation.
+ */
+function parseNumber(value: number | string, locale = navigator.language) {
```

Comment:
Let's swap the order of the parameters and turn the first parameter into an "options" object
```suggestion
function parseNumber(options: ParseNumberOptions = {locale: navigator.language}, value: number | string) {
```